azure-proto-core-test/SystemAssignedIdentityTests.cs

```diff
@@ -0,0 +1,348 @@
+                json = f.ReadToEnd();
+            JsonDocument document = JsonDocument.Parse(json);
+            JsonElement rootElement = document.RootElement;
+            var identityJsonProperty = rootElement.EnumerateObject().ElementAt(1);
+            SystemAssignedIdentity back = SystemAssignedIdentity.Deserialize(identityJsonProperty.Value);
+            Assert.IsTrue("de29bab1-49e1-4705-819b-4dfddceaaa98".Equals(back.PrincipalId.ToString()));
+            Assert.IsTrue("72f988bf-86f1-41af-91ab-2d7cd011db47".Equals(back.TenantId.ToString()));
+        }
+
+        [TestCase]
+        public void TestDeserializerBothValuesNull()
+        {
+            string json = "";
+            string path = ((AssemblyMetadataAttribute)GetType().Assembly.GetCustomAttribute(typeof(AssemblyMetadataAttribute))).Value;
+            using (StreamReader f = new StreamReader(path + "/TestAssets/SystemAssignedIdentity/SystemAssignedBothValuesNull.json"))
+                json = f.ReadToEnd();
+            JsonDocument document = JsonDocument.Parse(json);
+            JsonElement rootElement = document.RootElement;
+            var identityJsonProperty = rootElement.EnumerateObject().First();
+            var back = SystemAssignedIdentity.Deserialize(identityJsonProperty.Value);
+            Assert.IsNull(back);
+        }
+
+        [TestCase]
+        public void TestDeserializerBothEmptyString()
+        {
+            string json = "";
+            string path = ((AssemblyMetadataAttribute)GetType().Assembly.GetCustomAttribute(typeof(AssemblyMetadataAttribute))).Value;
+            using (StreamReader f = new StreamReader(path + "/TestAssets/SystemAssignedIdentity/SystemAssignedBothEmptyString.json"))
+                json = f.ReadToEnd();
+            JsonDocument document = JsonDocument.Parse(json);
+            JsonElement rootElement = document.RootElement;
+            var identityJsonProperty = rootElement.EnumerateObject().First();
+            Assert.Throws<FormatException>(delegate { Deserialize(identityJsonProperty.Value); });
+        }
+
+        [TestCase]
+        public void TestDeserializerOneEmptyString()
+        {
+            string json = "";
+            string path = ((AssemblyMetadataAttribute)GetType().Assembly.GetCustomAttribute(typeof(AssemblyMetadataAttribute))).Value;
+            using (StreamReader f = new StreamReader(path + "/TestAssets/SystemAssignedIdentity/SystemAssignedOneEmptyString.json"))
+                json = f.ReadToEnd();
+            JsonDocument document = JsonDocument.Parse(json);
+            JsonElement rootElement = document.RootElement;
+            var identityJsonProperty = rootElement.EnumerateObject().First();
+            Assert.Throws<FormatException>(delegate { Deserialize(identityJsonProperty.Value); });
+        }
+
+        [TestCase]
+        public void TestDeserializerTenantIdValueNull()
```

Comment:
these can be parameterized by passing in the path string since everything else appears the same